partition/src/partition.rs

```diff
@@ -16,13 +24,54 @@ impl ApexLinuxPartition {
         CONSTANTS.name.clone()
     }
 
+    #[cfg(feature = "socket")]
+    pub fn get_udp_socket(sockaddr: &str) -> Result<Option<UdpSocket>, ApexLinuxError> {
+        for stored in UDP_SOCKETS.iter() {
+            if stored.local_addr()?.to_string() == sockaddr {
+                let socket = stored.try_clone()?;
+                return Ok(Some(socket));
+            }
+        }
+        Ok(None)
+    }
+
+    #[cfg(feature = "socket")]
+    pub fn get_tcp_socket(sockaddr: &str) -> Result<Option<TcpStream>, ApexLinuxError> {
+        for stored in TCP_SOCKETS.iter() {
+            if stored.local_addr()?.to_string() == sockaddr {
+                let socket = stored.try_clone()?;
+                return Ok(Some(socket));
+            }
+        }
+        Ok(None)
+    }
+
     pub(crate) fn raise_system_error(error: SystemError) {
         if let Err(e) = SENDER.try_send(&PartitionCall::Error(error)) {
             panic!("Could not send SystemError event {error:?}. {e:?}")
         };
     }
 }
 
+#[derive(Debug, Clone)]
+pub enum ApexLinuxError {
+    SocketError,
```

Comment:
feature gate variant

Reply:
Feature-gating the single variant is not possible, only the full enum. I think this is because it is the only variant.